src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -3223,14 +3223,35 @@ private void CreateContainerProcessInternal()
                     // Create PowerShell process inside the container.
                     //
                     result = HcsCreateProcess(ComputeSystem, cmd, ref ProcessInformation, ref Process, ref resultString);
-                    if (result != 0)
+                    if (result == 0)
                     {
-                        processId = 0;
-                        error = result;
+                        processId = Convert.ToInt32(ProcessInformation.ProcessId);
+                    }
+                    else if (result == 2147942402) // 0x80070002 (2147942402) - The system cannot find the file specified.
+                    {
+                        // If we cannot find PowerShell Core, fall back to `powershell.exe` or Windows Powershell
+                        // We cannot do "cmd.exe /c powershell" because it returns process id of cmd instead powershell/pwsh.
+                        cmd = string.Format(System.Globalization.CultureInfo.InvariantCulture,
+                            @"{{""CommandLine"": ""powershell.exe {0} -NoLogo {1}"",""RestrictedToken"": {2}}}",
+                            (RuntimeId != Guid.Empty) ? "-so -NoProfile" : "-NamedPipeServerMode",
+                            String.IsNullOrEmpty(ConfigurationName) ? String.Empty : String.Concat("-Config ", ConfigurationName),
+                            (RunAsAdmin) ? "false" : "true");
+
+                        result = HcsCreateProcess(ComputeSystem, cmd, ref ProcessInformation, ref Process, ref resultString);
+                        if (result != 0)
+                        {
+                            processId = 0;
+                            error = result;
+                        }
+                        else
+                        {
+                            processId = Convert.ToInt32(ProcessInformation.ProcessId);
+                        }
                     }
                     else
                     {
-                        processId = Convert.ToInt32(ProcessInformation.ProcessId);
+                        processId = 0;
+                        error = result;
```

Comment:
I think you should set ErrorMessage here and in the failure path for failing to launch powershell.exe; otherwise, it won't be diagnosable (i.e., which path failed?  pwsh.exe or powershell.exe after not finding pwsh.exe)